apis/core/v1alpha1/adopted_resource.go

```diff
@@ -20,7 +20,7 @@ import (
 // AdoptedResourceSpec defines the desired state of the AdoptedResource.
 type AdoptedResourceSpec struct {
 	// +kubebuilder:validation:Required
-	Kubernetes *TargetKubernetesResource `json:"kubernetes"`
+	Kubernetes *AdoptedResourceTarget `json:"kubernetes"`
```

Comment:
It's just been renamed. The only thing that changes are the docstrings

Reply:
Existing AdoptedResource CRs in an existing installation of a controller would fail to be read into the controller, no? Since the over-wire format of the schema would be different...

Reply:
The types are still identical, and would share the same bit packing... I believe. Definitely something I could look into. I would like to think that changing the Go type name is irrelevant :(

Reply:
I reached out to Ellis about this:
> [Nick] I feel that if the CRD is identical, there shouldn't be a difference?
> [Ellis] The API Server serializes it using the CRD
> [Ellis] Exactly
> [Ellis] The APIServer isn't aware of your go struct, just the CRD
> [Ellis] We do this regularly in karpenter w/ our go types

So we don't need to worry about etcd serialization. Just that we are maintaining the same contract with the API server, which is the CRD. I have confirmed that the CRD remains untouched.